lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,252 @@
-        assert config.log_dir is None, f"Unexpected log directory, {config.log_dir!r}"
-        assert (
-            config.logger_type == "devlog"
-        ), f"Unexpected logger type, {config.logger_type!r}"
-        with pytest.raises(AttributeError):
-            print(f"{config.logger_host!r}")
-        with pytest.raises(AttributeError):
-            print(f"{config.logger_port!r}")
-        assert "42" == config.get(
-            "other", "foobar"
-        ), "Failed to fetch 'foobar' from 'DEFAULT' section"
-        assert "43" == config.get(
-            "other", "barfoo"
-        ), "Failed to fetch 'barfoo' from 'other' section"
-        assert isinstance(
-            config.files, list
-        ), f"Unexpected object class for 'files', {config.files.__class__!r}"
-
-    def test_log_dir_provided(self):
-        config = PbenchConfig(_config_path_prefix / "logdir.cfg")
-        assert (
-            config.log_dir == "/srv/log/directory"
-        ), f"Unexpected log directory, {config.log_dir!r}"
-
-    def test_logger_type_provided(self):
-        config = PbenchConfig(_config_path_prefix / "hostport.cfg")
-        assert (
-            config.logger_type == "hostport"
-        ), f"Unexpected logger type, {config.logger_type!r}"
-        assert (
-            config.logger_host == "logger.example.com"
-        ), f"Unexpected logger host value, {config.logger_host!r}"
-        assert (
-            config.logger_port == "42"
-        ), f"Unexpected logger port value, {config.logger_port!r}"
-
-    def test_logger_type_hostport_missing(self):
-        with pytest.raises(BadConfig):
-            PbenchConfig(_config_path_prefix / "hostport-missing.cfg")
-        with pytest.raises(BadConfig):
-            PbenchConfig(_config_path_prefix / "hostport-missing-port.cfg")
+    def mock_file_list(self, cfg_name: str) -> List[str]:
+        """file_list mock module"""
+        return [cfg_name]
+
+    def test_default_logger(self, monkeypatch):
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == "devlog"
+
+    def test_logger(self, monkeypatch):
+        """verify logger_type value from config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {"logging": {"logger_type": "file"}}
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == "file"
```

Comment:
I invite you to refactor this function like this:
```
    @staticmethod
    def logger_test_setup(sections: Dict[str, Any]):
        def mock_ConfigParser_init(self):
            self._sections = sections

        monkeypatch.setattr(configtools, "file_list", TestPbenchConfig.mock_file_list)
        monkeypatch.setattr(ConfigParser, "__init__", mock_ConfigParser_init)
        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)

    def test_logger(self, monkeypatch):
        """verify logger_type value from config file"""
        logger_type = "file"
        logger_test_setup({"logging": {"logger_type": logger_type}})
        config = PbenchConfig("pbench.cfg")
        assert config.logger_type == logger_type
```
Then, you can refactor `test_default_logger` similarly:
```
    def test_default_logger(self, monkeypatch):
        logger_test_setup({})
        config = PbenchConfig("pbench.cfg")
        assert config.logger_type == "devlog"
```
And then `test_logger_type_host_value`:
```
    def test_logger_type_host_value(self, monkeypatch):
        """verify `host` value of logger_type `hostport`"""
        logger_test_setup({"logging": {"logger_type": "hostport"}}}
        with pytest.raises(BadConfig) as exc:
            config = PbenchConfig("pbench.cfg")
            assert config.logger_type == "hostport"
            assert exc == "No option 'logger_host' in section: 'logging'"
```
And so on....